### PR TITLE
Plugin expenses

### PR DESCRIPTION
**Pull request by m1well**
### content
* track your expenses on an easy way and create a csv file (note) for further analysis 
* i also implemented an aggregate function, to aggregate the expenses of a year by month and cluster
* just read the README, there I have described it in detail
* I hope it works on my iPhone as good as on my MacBook ;)

### code
* hopefully the code fits in your "thoughts"
* I have also added some tests for my helper functions (functions without NP interactions)

### review
* just take your time with the review
* if I forgot something because I don't know much here yet, then please let me know
* if you have any questions, just ask in the discord
* what should I say: "it works on my machine" ;)

### CODE REVIEW

**Comment by mikeerickson** [comment]
@m1well I will have a look at this as soon as possible, but it likely won't be until tomorrow at the earliest as I am slammed with closing down work today.  Looking forward to seeing what you have done in your first attempt :-)

File: m1well.Expenses/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Expenses Plugin Changelog
+
+
+## [1.0.0] - 2021-11-24 (@m1well)
+
+### Added
```
Comment (mikeerickson) [inline m1well.Expenses/CHANGELOG.md:6]: You don't need to list added files for the first initial commit.  Surely not a show stopper but just food for thought.

File: m1well.Expenses/README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# Expenses Plugin
```
Comment (mikeerickson) [inline m1well.Expenses/README.md:1]: The first line of `README` should be in the form

# m1well.Expense Plugin

File: m1well.Expenses/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Expenses Plugin Changelog
```
Comment (mikeerickson) [inline m1well.Expenses/CHANGELOG.md:1]: The first line in `CHANGELOG` should be in the form

# m1well.Expenses Plugin Changelog

**Comment by mikeerickson** [comment]
@m1well All in all, this looks good.  I did not do extensive code analysis, I believe each plugin author should have the freedom to write code as they wish.  If there are issues in the future, I would be more inclined to dig deeper.  If you could just make the small formatting edits I have suggested to `README.md` and `CHANGELOG.md` that will suffice.

Note:  I would however like to see some tests added to the project

**Comment by m1well** [comment]
all changes completed ;)

**Review by mikeerickson — APPROVED** [review]

**Pull request merged**